### PR TITLE
Clarify different federated schema types

### PR DESCRIPTION
**Pull request by StephenBarlow**

### CODE REVIEW

File: docs/source/entities.md [hunk 1]
```diff
@@ -3,10 +3,6 @@ title: Entities
 description: Reference and extend types across subgraphs
 ---
 
-> **Implementing services** are now known as **subgraphs**.
```
Comment (StephenBarlow) [inline docs/source/entities.md:6]: Cutting these becasue they've been up long enough

File: docs/source/index.mdx [hunk 2]
```diff
@@ -5,14 +5,13 @@ description: Implement a single data graph across multiple services
...
->
-> If you find an outdated use of "implementing service," please submit article feedback or a pull request using the links on the right.
+import {
+  ExpansionPanel,
+} from 'gatsby-theme-apollo-docs/src/components/expansion-panel';
 
 To get the most out of GraphQL, your organization should expose a [single data graph](https://principledgraphql.com/integrity#1-one-graph) that provides a unified interface for querying any combination of your backing data sources. However, it can be challenging to represent an enterprise-scale data graph with a single, monolithic GraphQL server.
 
-To remedy this, you can divide your graph's implementation across multiple composable services with **Apollo Federation**:
+To remedy this, you can divide your graph's implementation across multiple back-end services with **Apollo Federation**:
```
Comment (trevor-scheer) [inline docs/source/index.mdx:14]: Possibly? I think what you have is also good, just proposing another option. Subgraph might be unintroduced jargon just yet.

```suggestion
To remedy this, you can divide your graph's implementation across multiple subgraphs with **Apollo Federation**:
```

File: docs/source/index.mdx [hunk 3]
```diff
@@ -76,18 +75,20 @@ The difference is that now, each subgraph defines the types and fields that it i
...
 
-## Apollo Server libraries
+## Federated schemas
 
-Apollo Server supports Apollo Federation via two open-source extension libraries:
+A federated graph uses multiple "types" of GraphQL schemas:
 
-* `@apollo/federation` provides primitives that subgraphs use to make their individual GraphQL schemas composable.
-* `@apollo/gateway` enables you to set up an instance of Apollo Server as a **gateway** that distributes incoming GraphQL operations across one or more  subgraphs.
+* **Subgraph schemas.** Each of your **subgraphs** has its own schema that represents the types and fields of your **supergraph** that it's responsible for resolving.
```
Comment (trevor-scheer) [inline docs/source/index.mdx:82]: I think this needs just a just a bit more clarity. Might even add one more statement that says collectively, the subgraphs must be capable of resolving the entire supergraph.

Edit: the addition I'm proposing _is_ mentioned immediately below, so maybe disregard.

```suggestion
* **Subgraph schemas.** Each of your **subgraphs** has its own schema that represents some of the types and fields of your **supergraph** that it's responsible for resolving.
```

File: docs/source/index.mdx [hunk 4]
```diff
@@ -121,6 +122,7 @@ type Review {
 
 extend type User @key(fields: "id") {
   id: ID! @external
+  username: String! @external
```
Comment (trevor-scheer) [inline docs/source/index.mdx:125]: This is inappropriately marked as external. Only fields that are referenced by a key, requires, or provides should be marked external.
https://github.com/apollographql/federation/blob/c3bd6ddca9ed148c71eddb7969a5030148f3b582/docs/source/errors.md#L35

File: docs/source/index.mdx [hunk 5]
```diff
@@ -140,32 +142,139 @@ _another_ subgraph. For example, the `Review` type includes a
...
+}
+```
+
+</ExpansionPanel>
 
 ## Gateway example
 
-The gateway for our federated data graph fetches the schema from each
-subgraph and composes those schemas into a single graph:
+The gateway for a federated data graph provides its composed [supergraph schema](#supergraph-schema) to the `ApolloGateway` constructor:
```
Comment (trevor-scheer) [inline docs/source/index.mdx:262]: Reads a little funny, and I'm not sure what actually goes in \<here\>
>"The \<gateway\>...provides its supergraph schema to the `ApolloGateway` constructor"

"The {node|server} process", possibly?

File: docs/source/index.mdx [hunk 6]
```diff
@@ -140,32 +142,139 @@ _another_ subgraph. For example, the `Review` type includes a
...
+
+```js{4}
+const supergraphSchema = readFileSync('./supergraph.graphql').toString();
```
Comment (trevor-scheer) [inline docs/source/index.mdx:265]: Nit: rename this variable to `supergraphSdl` and shorthand it below. In code, `sdl` and `schema` do have somewhat distinct meanings, and this is definitely sdl.

**Review by trevor-scheer — CHANGES_REQUESTED** [review]

File: docs/source/index.mdx (hunk 2 above)
Comment (trevor-scheer) [reply docs/source/index.mdx:14]: The "possibly?" is commentary directed at my own suggestion, btw.

Comment (StephenBarlow) [reply docs/source/index.mdx:14]: It is just barely unintroduced, but I will fix by introducing just-in-time 😄

File: docs/source/index.mdx (hunk 4 above)
Comment (StephenBarlow) [reply docs/source/index.mdx:125]: Ah, so, `username` _is_ used in a `@provides`, but up in the `Review` type just above here. If I try to compose these exact subgraphs w/ Rover and _don't_ include `username` here with `@external`, I get a composition error.

Comment (trevor-scheer) [reply docs/source/index.mdx:136]: Ohhh, then great catch! At the glance I took, I wasn't even aware of the review type / that the SDL was collapsed 👍

**Comment by trevor-scheer** [comment]
🙌 approved in post!

**Pull request merged**